core/src/main/java/me/lokka30/treasury/plugin/core/utils/UpdateChecker.java

```diff
@@ -43,21 +46,75 @@ public static void checkForUpdates() {
         if (!plugin.configAdapter().getSettings().checkForUpdates()) {
             return;
         }
-        plugin.scheduler().runAsync(() -> {
-            try {
-                URL url = new URL(REQUEST_URI);
-                try (Reader in = new InputStreamReader(url.openStream())) {
-                    JsonObject object = Utils.GSON.fromJson(in, JsonObject.class);
-                    String latestVersion = object.get("current_version").getAsString();
-                    handleVersionCheck(latestVersion);
+        // since spigot is going to get only releases, we shall check whether this version is a
+        // release first and if it is a release, check spigot, otherwise check github commits
+        PluginVersion current = plugin.getVersion();
+        if (current.isReleaseVersion()) {
+            plugin.scheduler().runAsync(() -> {
+                try {
+                    URL url = new URL(SPIGOT_REQUEST_URI);
+                    try (Reader in = new InputStreamReader(url.openStream())) {
+                        JsonObject object = Utils.GSON.fromJson(in, JsonObject.class);
+                        String latestVersion = object.get("current_version").getAsString();
+                        handleSpigotVersionCheck(latestVersion);
+                    }
+                } catch (IOException e) {
+                    plugin.logger().error("IO whilst trying to request Spigot", e);
```

Comment:
This message will be seem by server owners, they may not even know what IO is. Maybe a more user friendly message is "Exception while getting plugin data from spigot"

Reply:
Suggestion: `Unable to retrieve the latest version data for the Treasury resource on SpigotMC.org. Is your internet connection experiencing any trouble?`